VMEncryption/main/test/test_check_util.py

```diff
@@ -57,6 +57,22 @@ def test_is_kv_id(self):
         self.assertRaises(Exception, self.cutil.check_kv_id, "/subscriptions/{subid}/resourceGroupssss/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}", "")
         self.assertRaises(Exception, self.cutil.check_kv_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}", "")
 
+    def test_is_mhsm_id(self):
+         # https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules
+        self.cutil.check_mhsm_id("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/adenszqtrrg/providers/Microsoft.KeyVault/managedHSM/adenszqtrkv", "")
+        self.cutil.check_mhsm_id("/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/adenszqtrrg/providers/microsoft.keyvault/managedHSM/adenszqtrkv", "")
+
+        # invalid cases
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "////", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault////////", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroupssss/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/vaults/{vaultname}", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroupssss/{rgname}/providers/Microsoft.KeyVault/managedHSM/{vaultname}", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.KeyVault/managedHSM/{vaultname}", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.ManagedHSM/managedHSM/{vaultname}", "")
+        self.assertRaises(Exception, self.cutil.check_mhsm_id, "/subscriptions/{subid}/resourceGroups/{rgname}/providers/Microsoft.ManagedHSM/vaults/{vaultname}", "")
+
```

Comment:
Why can't we just have some test names for subid (guid of subscriptionId) , rgname (something like : testrg) and valutname (like : testkv) instead of generic names such as {subid} , {rgname} and {vaultname} ? The tests may become slightly confusing because we expect the proper names for these parameters and not the one surrounded with `{` and `}`.

Reply:
I copied from the kv_id checks. My guess is that that is left over sometimes from templates/docs, so it is a test to see if bad parameters are caught? This test expects failure. There are also other tests without the {}

Reply:
Oh got it. Then nevermind . I did not see that the its a negative test.